<a href="https://colab.research.google.com/github/fardsnog2/ML_vvsu/blob/main/lab1/%D0%97%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Задание по практике 1

##Загрузка датасета
В файле classification datasets расположены ссылки на датасеты. Выберите один в соответствии со своим вариантом, скачайте и прочитайте файл

**Не забудьте загрузить нужные библиотеки**

Например: pip3 install pandas

In [1]:
import pandas as pd

In [10]:
df = pd.read_excel('/content/moviedb.xls')
df=df.drop(columns=['Unnamed: 0'])
df

title              release  \
0                                       "Vegeta" (Shrek)                  NaN   
1                                          '93: Del Soul  2021-07-18 00:00:00   
2               'Catch Me If You Can': Behind the Camera  2003-05-06 00:00:00   
3                       'Harry Potter': Behind the Magic  2005-11-19 00:00:00   
4      'Oh, Miss Jones!': The Very Best of Leonard Ro...  1996-08-16 00:00:00   
...                                                  ...                  ...   
10277                                            Белфаст  2021-11-12 00:00:00   
10278                                               Финч  2021-11-04 00:00:00   
10279                                       Под гипнозом  2021-10-27 00:00:00   
10280                                    Последняя дуэль  2021-10-13 00:00:00   
10281                                Параллельные матери  2021-10-08 00:00:00   

       runtime       budget      revenue lang          genre1   genre2  \
0         99.0          NaN          NaN  NaN             NaN      NaN   
1         15.0        250.0          NaN   en        детектив  триллер   
2         17.0          NaN          NaN  NaN  документальный      NaN   
3         48.0          NaN          NaN  NaN             NaN      NaN   
4         52.0          NaN          NaN   en  документальный      NaN   
...        ...          ...          ...  ...             ...      ...   
10277     98.0          NaN   30000000.0   en           драма      NaN   
10278    115.0    2500000.0  500000000.0   en      фантастика    драма   
10279     88.0          NaN          NaN   en         триллер    драма   
10280    152.0  100000000.0   27000000.0   en          боевик    драма   
10281    123.0          NaN          NaN   es           драма      NaN   

      country1 country2  ...            writer                  studio1  \
0          NaN      NaN  ...               NaN                      NaN   
1          NaN      NaN  ...    Riley Lockhart       Elkman Productions   
2          NaN      NaN  ...               NaN                      NaN   
3          NaN      NaN  ...               NaN                      NaN   
4           GB      NaN  ...               NaN                 Telegram   
...        ...      ...  ...               ...                      ...   
10277       GB      NaN  ...   Kenneth Branagh  Northern Ireland Screen   
10278       US      NaN  ...       Ivor Powell     Amblin Entertainment   
10279       CA      NaN  ...  Richard D'Ovidio            The Long Game   
10280       GB       US  ...               NaN     20th Century Studios   
10281       ES      NaN  ...               NaN                 El Deseo   

                      studio2                keyword1              keyword2  \
0                         NaN                     NaN                   NaN   
1                         NaN              sports car  psychological stress   
2                         NaN                     NaN                   NaN   
3                         NaN                     NaN                   NaN   
4                         NaN                     NaN                   NaN   
...                       ...                     ...                   ...   
10277                    TKBC           working class       black and white   
10278             ImageMovers                   robot               journey   
10279                     NaN                     NaN                   NaN   
10280  Scott Free Productions  based on novel or book      normandy, france   
10281                     TVE                  mother          photographer   

                     keyword3           keyword4           keyword5  \
0                         NaN                NaN                NaN   
1                  axe murder  horror filmmaking              1990s   
2                         NaN                NaN                NaN   
3                         NaN                NaN       

##Первое знакомство с датасетом
В следующих ячейках выведите:


1.   Колонки датасета
2.   Первые 5 значений
3.   Последние 5 значений
4.   Статистику по числовым колонкам



In [11]:
df.columns

Index(['title', 'release', 'runtime', 'budget', 'revenue', 'lang', 'genre1',
       'genre2', 'country1', 'country2', 'actor1', 'actor2', 'actor3',
       'actor4', 'actor5', 'director', 'writer', 'studio1', 'studio2',
       'keyword1', 'keyword2', 'keyword3', 'keyword4', 'keyword5', 'keyword6',
       'rating'],
      dtype='object')

In [12]:
df.head(5)

title              release  \
0                                   "Vegeta" (Shrek)                  NaN   
1                                      '93: Del Soul  2021-07-18 00:00:00   
2           'Catch Me If You Can': Behind the Camera  2003-05-06 00:00:00   
3                   'Harry Potter': Behind the Magic  2005-11-19 00:00:00   
4  'Oh, Miss Jones!': The Very Best of Leonard Ro...  1996-08-16 00:00:00   

   runtime  budget  revenue lang          genre1   genre2 country1 country2  \
0     99.0     NaN      NaN  NaN             NaN      NaN      NaN      NaN   
1     15.0   250.0      NaN   en        детектив  триллер      NaN      NaN   
2     17.0     NaN      NaN  NaN  документальный      NaN      NaN      NaN   
3     48.0     NaN      NaN  NaN             NaN      NaN      NaN      NaN   
4     52.0     NaN      NaN   en  документальный      NaN       GB      NaN   

   ...          writer             studio1 studio2    keyword1  \
0  ...             NaN                 NaN     NaN         NaN   
1  ...  Riley Lockhart  Elkman Productions     NaN  sports car   
2  ...             NaN                 NaN     NaN         NaN   
3  ...             NaN                 NaN     NaN         NaN   
4  ...             NaN            Telegram     NaN         NaN   

               keyword2    keyword3           keyword4 keyword5  \
0                   NaN         NaN                NaN      NaN   
1  psychological stress  axe murder  horror filmmaking    1990s   
2                   NaN         NaN                NaN      NaN   
3                   NaN         NaN                NaN      NaN   
4                   NaN         NaN                NaN      NaN   

         keyword6 rating  
0             NaN    0.0  
1  carsploitation    2.0  
2             NaN    8.7  
3             NaN    8.0  
4             NaN    6.0  

[5 rows x 26 columns]

In [13]:
df.head(-5)

title              release  \
0                                       "Vegeta" (Shrek)                  NaN   
1                                          '93: Del Soul  2021-07-18 00:00:00   
2               'Catch Me If You Can': Behind the Camera  2003-05-06 00:00:00   
3                       'Harry Potter': Behind the Magic  2005-11-19 00:00:00   
4      'Oh, Miss Jones!': The Very Best of Leonard Ro...  1996-08-16 00:00:00   
...                                                  ...                  ...   
10272                                         The Father  2021-03-21 00:00:00   
10273                                       По наклонной  2021-02-26 00:00:00   
10274                                          Чернобыль  2021-04-15 00:00:00   
10275                                            Часовой  2021-03-05 00:00:00   
10276                                          Дом Gucci  2021-11-24 00:00:00   

       runtime      budget      revenue lang          genre1    genre2  \
0         99.0         NaN          NaN  NaN             NaN       NaN   
1         15.0       250.0          NaN   en        детектив   триллер   
2         17.0         NaN          NaN  NaN  документальный       NaN   
3         48.0         NaN          NaN  NaN             NaN       NaN   
4         52.0         NaN          NaN   en  документальный       NaN   
...        ...         ...          ...  ...             ...       ...   
10272      5.0         NaN          NaN   en           драма       NaN   
10273    142.0  40000000.0          NaN   en           драма  криминал   
10274    120.0   9100000.0    6459306.0   ru           драма   история   
10275     80.0   6500000.0          NaN   fr         триллер    боевик   
10276    158.0  75000000.0  151758425.0   en           драма  криминал   

      country1 country2  ...           writer                 studio1  \
0          NaN      NaN  ...              NaN                     NaN   
1          NaN      NaN  ...   Riley Lockhart      Elkman Productions   
2          NaN      NaN  ...              NaN                     NaN   
3          NaN      NaN  ...              NaN                     NaN   
4           GB      NaN  ...              NaN                Telegram   
...        ...      ...  ...              ...                     ...   
10272       GB      NaN  ...     Blake Ridder            Ridder Films   
10273       US      NaN  ...              NaN                    AGBO   
10274       RU      NaN  ...              NaN    Non-Stop Productions   
10275       FR      NaN  ...  Julien Leclercq        Labyrinthe Films   
10276       CA       GB  ...              NaN  Scott Free Productions   

                          studio2                        keyword1  \
0                             NaN                             NaN   
1                             NaN                      sports car   
2                             NaN                             NaN   
3                             NaN                             NaN   
4                             NaN                             NaN   
...                           ...                             ...   
10272                         NaN                             NaN   
10273  The Hideaway Entertainment  post traumatic stress disorder   
10274            DK Entertainment                          diving   
10275                         NaN  post traumatic stress disorder   
10276         Metro-Goldwyn-Mayer                       biography   

                     keyword2             keyword3           keyword4  \
0                         NaN                  NaN                NaN   
1        psychological stress           axe murder  horror filmmaking   
2                         NaN                  NaN                NaN   
3                         NaN                  NaN                NaN   
4                         NaN                  NaN                NaN   
...                       ...                

In [14]:
df.describe()

runtime        budget       revenue        rating
count  10130.000000  4.119000e+03  4.113000e+03  10240.000000
mean      88.593287  3.986407e+07  1.177812e+08      5.181666
std       40.281324  4.822654e+07  2.064450e+08      2.599802
min        0.000000  1.000000e+00  1.000000e+00      0.000000
25%       83.000000  7.500000e+06  9.128263e+06      4.900000
50%       96.000000  2.200000e+07  4.266441e+07      6.100000
75%      109.000000  5.200000e+07  1.293948e+08      6.800000
max      840.000000  3.800000e+08  2.797801e+09     10.000000

**Ответьте на следующие вопросы:**


1.   Какие средние значения в каждой числовой колонке вашего датасета?
2.   Минимальные и максимальные значения в каждой числовой колонке вашего датасета





1.   **Ответ 1.**5140.500000,	88.593287,	3.986407e+07,	1.177812e+08,	5.181666
2.   **Ответ 2.** **min** 0.000000,	0.000000,	1.000000e+00,	1.000000e+00,	0.000000
**max**10281.000000,	840.000000,	3.800000e+08,	2.797801e+09,	10.000000


##Типизация
Присвойте соответствующим колонкам один из типов (category, float, int)

In [21]:
df=df.dropna()
df['rating']=df['rating'].astype("float")
df['runtime']=df['runtime'].astype("int")
df['genre1']=df['genre1'].astype("category")
df.info()
#Вывести типы колонок тут

<class 'pandas.core.frame.DataFrame'>
Int64Index: 660 entries, 8 to 10242
Data columns (total 26 columns):
 #   Column    Non-Null Count  Dtype   
---  ------    --------------  -----   
 0   title     660 non-null    object  
 1   release   660 non-null    object  
 2   runtime   660 non-null    int64   
 3   budget    660 non-null    float64 
 4   revenue   660 non-null    float64 
 5   lang      660 non-null    object  
 6   genre1    660 non-null    category
 7   genre2    660 non-null    object  
 8   country1  660 non-null    object  
 9   country2  660 non-null    object  
 10  actor1    660 non-null    object  
 11  actor2    660 non-null    object  
 12  actor3    660 non-null    object  
 13  actor4    660 non-null    object  
 14  actor5    660 non-null    object  
 15  director  660 non-null    object  
 16  writer    660 non-null    object  
 17  studio1   660 non-null    object  
 18  studio2   660 non-null    object  
 19  keyword1  660 non-null    object  
 20  keyword2

Какие признаки вы можете выделить в своем датасете (категориальные, числовые и др., **минимум 3 разных**, помните что можно также выделить признаки более точно, например числовые - порядковые, количественные и т.д.)?
**ОТВЕТ** числовые - рейтинг-кол-ые, бюджет-кол-ые
категориальные - жанры
релиз - временные

##Фильтрация
Офильтруйте 3 колонки вашего датасета и выведите результат

In [26]:

df.loc[df['rating'] >7 ]

title              release  runtime  \
9                         007: Казино Рояль                39035      139   
11               007: Координаты «Скайфолл»                41207      143   
33                           12 лет рабства                41565      134   
39                                 13 убийц                40430      141   
1593               Kingsman: Золотое кольцо                42998      141   
...                                     ...                  ...      ...   
9985                                 Джокер  2019-10-02 00:00:00      122   
10081                        Вход в пустоту  2009-06-17 00:00:00      161   
10122                           Книга жизни  2014-10-01 00:00:00       95   
10232  Охотники за привидениями: Наследники  2021-11-11 00:00:00      124   
10242                          Игры разумов  2019-03-07 00:00:00      124   

            budget       revenue lang       genre1       genre2 country1  \
9      150000000.0  5.990460e+08   en  приключения       боевик       DE   
11     200000000.0  1.108561e+09   en       боевик  приключения       GB   
33      20000000.0  1.870000e+08   en        драма      история       US   
39       6000000.0  1.755514e+07   ja  приключения        драма       JP   
1593   104000000.0  4.109027e+08   sv       боевик  приключения       GB   
...            ...           ...  ...          ...          ...      ...   
9985    55000000.0  1.074251e+09   en     криминал      триллер       CA   
10081   13000000.0  7.542490e+05   en      фэнтези        драма       BR   
10122   50000000.0  9.743711e+07   en   мультфильм  приключения       MX   
10232   75000000.0  1.910000e+08   en      фэнтези      комедия       CA   
10242   25000000.0  4.957273e+06   en      история        драма       IE   

      country2  ...                writer                   studio1  \
9           US  ...       Eon Productions           Stillking Films   
11          US  ...                Danjaq         Columbia Pictures   
33          GB  ...  New Regency Pictures      Plan B Entertainment   
39          GB  ...        Daisuke Tengan        Toho Company, Ltd.   
1593        US  ...        Matthew Vaughn           2th Century Fox   
...        ...  ...                   ...                       ...   
9985        US  ...          Scott Silver     Warner Bros. Pictures   
10081       CA  ...            Gaspar Noé         BIM Distribuzione   
10122       US  ...         Doug Langdale  Reel FX Creative Studios   
10232       US  ...         Jason Reitman         Columbia Pictures   
10242       US  ...        Farhad Safinia             Fastnet Films   

                   studio2                keyword1  \
9                    italy                   poker   
11                     spy            secret agent   
33                 slavery              plantation   
39     Sedic International                   japan   
1593            Marv Films        saving the world   
...                    ...                     ...   
9985          Joint Effort             street gang   
10081           Wild Bunch              prostitute   
10122             Chatrone           love triangle   
10232         Bron Studios              small town   
10242                22h22  based on novel or book   

                               keyword2                keyword3  \
9                                casino  based on novel or book   
11                            sociopath                  killer   
33     based on memoir or autobiography           violin player   
39                              suicide                assassin   
1593                                spy                  europe   
...                                 ...                     ...   
9985                              dream                   clown   
10081                     hallucination              strip club   
10122                            mexico               afterlife   
10232                       

In [27]:
df.loc[df['budget'] >1000000 ]

title              release  runtime  \
8              007: Завтра не умрёт никогда  1997-12-11 00:00:00      119   
9                         007: Казино Рояль                39035      139   
11               007: Координаты «Скайфолл»                41207      143   
12                       007: Лунный гонщик  1979-06-26 00:00:00      126   
13         007: Никогда не говори «никогда»  1983-10-07 00:00:00      134   
...                                     ...                  ...      ...   
10122                           Книга жизни  2014-10-01 00:00:00       95   
10129                                Ведьма  2015-01-27 00:00:00       92   
10149             Человек – швейцарский нож  2016-06-24 00:00:00       95   
10232  Охотники за привидениями: Наследники  2021-11-11 00:00:00      124   
10242                          Игры разумов  2019-03-07 00:00:00      124   

            budget       revenue lang       genre1       genre2 country1  \
8      110000000.0  3.330111e+08   de  приключения       боевик       GB   
9      150000000.0  5.990460e+08   en  приключения       боевик       DE   
11     200000000.0  1.108561e+09   en       боевик  приключения       GB   
12      34000000.0  2.103081e+08   en       боевик  приключения       FR   
13      36000000.0  1.600000e+08   en  приключения       боевик       DE   
...            ...           ...  ...          ...          ...      ...   
10122   50000000.0  9.743711e+07   en   мультфильм  приключения       MX   
10129    3500000.0  4.042394e+07   en        драма      фэнтези       CA   
10149    3000000.0  4.210454e+06   en      комедия        драма       SE   
10232   75000000.0  1.910000e+08   en      фэнтези      комедия       CA   
10242   25000000.0  4.957273e+06   en      история        драма       IE   

      country2  ...             writer                   studio1  \
8           US  ...    Eon Productions            United Artists   
9           US  ...    Eon Productions           Stillking Films   
11          US  ...             Danjaq         Columbia Pictures   
12          GB  ...     United Artists           Eon Productions   
13          GB  ...  PSO International  TaliaFilm II Productions   
...        ...  ...                ...                       ...   
10122       US  ...      Doug Langdale  Reel FX Creative Studios   
10129       GB  ...      Robert Eggers     Very Special Projects   
10149       US  ...   Daniel Scheinert                 BlackBird   
10232       US  ...      Jason Reitman         Columbia Pictures   
10242       US  ...     Farhad Safinia             Fastnet Films   

               studio2                keyword1        keyword2  \
8      london, england                 england             spy   
9                italy                   poker          casino   
11                 spy            secret agent       sociopath   
12       venice, italy             mass murder  rio de janeiro   
13          video game                     spy         bahamas   
...                ...                     ...             ...   
10122         Chatrone           love triangle          mexico   
10129  Parts and Labor                   witch       isolation   
10149           Tadmor                   beach         friends   
10232     Bron Studios              small town     ghostbuster   
10242            22h22  based on novel or book       professor   

                     keyword3         keyword4                keyword5  \
8                       china     intelligence              television   
9      based on novel or book        terrorist                  banker   
11                     killer      art gallery  british secret service   
12               space marine       space suit           marcus square   
13                   stealing     scuba diving                   scuba   
...                       ...              ...                     ...   
10122               afterlife  day of the dead            bullfighting  

In [28]:
df.loc[df['genre1'] =='приключения' ]

title              release  runtime  \
8         007: Завтра не умрёт никогда  1997-12-11 00:00:00      119   
9                    007: Казино Рояль                39035      139   
13    007: Никогда не говори «никогда»  1983-10-07 00:00:00      134   
16          007: Только для твоих глаз  1981-06-23 00:00:00      127   
17             007: Умри, но не сейчас                37577      133   
...                                ...                  ...      ...   
9207                         Центурион                40224       97   
9217        Чарли и шоколадная фабрика                38546      115   
9345              Чужой против Хищника                38211      100   
9378                Шанхайский полдень                36672      110   
9387          Шерлок Холмс: Игра теней                40869      128   

           budget      revenue lang       genre1      genre2 country1  \
8     110000000.0  333011068.0   de  приключения      боевик       GB   
9     150000000.0  599045960.0   en  приключения      боевик       DE   
13     36000000.0  160000000.0   en  приключения      боевик       DE   
16     28000000.0  195312802.0   el  приключения      боевик       GB   
17    140000000.0  431971116.0   en  приключения      боевик       GB   
...           ...          ...  ...          ...         ...      ...   
9207   12000000.0    6890432.0   gd  приключения      боевик       FR   
9217  150000000.0  474968763.0   en  приключения     комедия       AU   
9345   70000000.0  177427090.0   it  приключения  фантастика       CA   
9378   55000000.0   56932305.0   en  приключения      боевик       HK   
9387  125000000.0  334615000.0   en  приключения      боевик       GB   

     country2  ...                      writer                       studio1  \
8          US  ...             Eon Productions                United Artists   
9          US  ...             Eon Productions               Stillking Films   
13         GB  ...           PSO International      TaliaFilm II Productions   
16         US  ...              United Artists               Eon Productions   
17         US  ...             Eon Productions           Metro-Goldwyn-Mayer   
...       ...  ...                         ...                           ...   
9207       GB  ...               Neil Marshall  Pathé Pictures International   
9217       GB  ...   Village Roadshow Pictures            The Zanuck Company   
9345       CZ  ...      Revolution Sun Studios               Impact Pictures   
9378       US  ...  Roger Birnbaum Productions        Spyglass Entertainment   
9387       US  ...       Warner Bros. Pictures     Village Roadshow Pictures   

               studio2         keyword1             keyword2  \
8      london, england          england                  spy   
9                italy            poker               casino   
13          video game              spy              bahamas   
16     london, england        submarine              england   
17                 spy            laser            espionage   
...                ...              ...                  ...   
9207   UK Film Council     roman empire         ancient rome   
9217   london, england        chocolate       factory worker   
9345  saving the world         predator         space marine   
9378      martial arts         princess                sioux   
9387     paris, france  london, england  detective inspector   

                    keyword3                   keyword4  \
8                      china               intelligence   
9     based on novel or book                  terrorist   
13                  stealing               scuba diving   
16                       sea                   assassin   
17    british secret service               havana, cuba   
...                      ...                        ...   
9207           ancient world                   violence   
9217  based on novel or book  parent child relationship   
9345                 pyramid  

##Группировка
Сгруппируйте датасет по какому-либо категориальному признаку и выведите среднее значение численных колонок (см. метод mean в pandas)

In [33]:
df.groupby('genre1').mean()

<ipython-input-33-812ccae3191a>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df.groupby('genre1').mean()


runtime        budget       revenue    rating
genre1                                                       
боевик       113.734177  7.546837e+07  2.159428e+08  6.406256
вестерн      138.000000  8.366667e+07  2.148323e+08  7.042300
военный      122.285714  7.042857e+07  1.521988e+08  6.957143
детектив     115.750000  5.575000e+07  9.379681e+07  5.625000
драма        119.283784  2.839035e+07  6.705882e+07  6.895040
история      122.600000  2.520000e+07  1.268594e+08  7.100000
комедия      104.048387  2.956658e+07  9.781789e+07  6.572462
криминал     116.025000  2.781462e+07  9.037720e+07  6.774551
мелодрама    121.250000  2.864167e+07  7.973866e+07  6.707970
музыка       127.500000  4.600000e+07  5.306757e+08  7.677188
мультфильм    88.357143  5.471429e+07  1.505785e+08  6.557500
приключения  121.486842  9.705713e+07  2.977244e+08  6.594004
семейный      99.250000  5.418750e+07  1.406195e+08  6.550000
триллер      110.194444  4.073472e+07  9.557122e+07  6.481241
ужасы         99.673913  2.511170e+07  6.771044e+07  6.079388
фантастика   114.333333  8.119444e+07  2.285518e+08  6.685958
фэнтези      115.904762  6.003573e+07  1.539705e+08  6.566246

##Задание для гениев *
Агрегируйте числовой признак по категориальному признаку с использованием метода **без использования** библиотеки pandas, и вычислите медиану и среднее значение для каждой категории.

In [34]:
df.groupby('genre1').agg(['median', 'mean'])

<ipython-input-34-8de10bd62461>:1: FutureWarning: ['title', 'release', 'lang', 'genre2', 'country1', 'country2', 'actor1', 'actor2', 'actor3', 'actor4', 'actor5', 'director', 'writer', 'studio1', 'studio2', 'keyword1', 'keyword2', 'keyword3', 'keyword4', 'keyword5', 'keyword6'] did not aggregate successfully. If any error is raised this will raise in a future version of pandas. Drop these columns/ops to avoid this warning.
  df.groupby('genre1').agg(['median', 'mean'])


runtime                   budget                    revenue  \
             median        mean       median          mean       median   
genre1                                                                    
боевик        110.5  113.734177   57000000.0  7.546837e+07  122624829.0   
вестерн       136.0  138.000000  100000000.0  8.366667e+07  108103450.0   
военный       114.0  122.285714   68000000.0  7.042857e+07   94882889.0   
детектив      114.0  115.750000   55000000.0  5.575000e+07   58180297.0   
драма         118.5  119.283784   16750000.0  2.839035e+07   34720936.5   
история       121.0  122.600000   14000000.0  2.520000e+07  115890699.0   
комедия       101.0  104.048387   23000000.0  2.956658e+07   47360245.5   
криминал      114.5  116.025000   20000000.0  2.781462e+07   34486511.5   
мелодрама     117.0  121.250000   19200000.0  2.864167e+07   19043658.0   
музыка        127.5  127.500000   46000000.0  4.600000e+07  530675746.5   
мультфильм     87.0   88.357143   50000000.0  5.471429e+07  124800279.5   
приключения   119.0  121.486842   94000000.0  9.705713e+07  207677684.0   
семейный       92.0   99.250000   46750000.0  5.418750e+07  108668465.5   
триллер       109.0  110.194444   30000000.0  4.073472e+07   54204701.5   
ужасы          95.5   99.673913   16000000.0  2.511170e+07   52432091.0   
фантастика    113.5  114.333333   75000000.0  8.119444e+07  134328919.5   
фэнтези       108.0  115.904762   50000000.0  6.003573e+07   95396573.0   

                             rating            
                     mean    median      mean  
genre1                                         
боевик       2.159428e+08  6.300000  6.406256  
вестерн      2.148323e+08  6.800000  7.042300  
военный      1.521988e+08  6.800000  6.957143  
детектив     9.379681e+07  5.900000  5.625000  
драма        6.705882e+07  7.000000  6.895040  
история      1.268594e+08  7.100000  7.100000  
комедия      9.781789e+07  6.550000  6.572462  
криминал     9.037720e+07  6.650000  6.774551  
мелодрама    7.973866e+07  6.850000  6.707970  
музыка       5.306757e+08  7.677188  7.677188  
мультфильм   1.505785e+08  6.300000  6.557500  
приключения  2.977244e+08  6.540387  6.594004  
семейный     1.406195e+08  6.650000  6.550000  
триллер      9.557122e+07  6.600000  6.481241  
ужасы        6.771044e+07  6.200000  6.079388  
фантастика   2.285518e+08  6.747949  6.685958  
фэнтези      1.539705e+08  6.600000  6.566246